analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.cs

```diff
@@ -24,1068 +24,1109 @@
+    private record Expected(string[] Names, ExpectedKind Kind);
 
-        private record Expected(string[] Names, ExpectedKind Kind);
+    private class Context
+    {
+        public readonly RoslynLiveVariableAnalysis Lva;
+        public readonly ControlFlowGraph Cfg;
 
-        private class Context
+        public Context(string code, AnalyzerLanguage language, string localFunctionName = null)
         {
-            public readonly RoslynLiveVariableAnalysis Lva;
-            public readonly ControlFlowGraph Cfg;
+            Cfg = TestHelper.CompileCfg(code, language, code.Contains("// Error CS"), localFunctionName);
+            Lva = new RoslynLiveVariableAnalysis(Cfg, default);
+        }
 
-            public Context(string code, AnalyzerLanguage language, string localFunctionName = null)
-            {
-                Cfg = TestHelper.CompileCfg(code, language, code.Contains("// Error CS"), localFunctionName);
-                Lva = new RoslynLiveVariableAnalysis(Cfg, default);
-            }
+        public Context(string code, SyntaxKind syntaxKind)
+        {
+            var (tree, model) = TestHelper.Compile(code, false, AnalyzerLanguage.CSharp);
+            var node = tree.GetRoot().DescendantNodes().First(x => x.RawKind == (int)syntaxKind);
+            Cfg = node.CreateCfg(model, default);
+            Lva = new RoslynLiveVariableAnalysis(Cfg, default);
+        }
 
-            public Context(string code, SyntaxKind syntaxKind)
+        public void ValidateAllEmpty()
+        {
+            foreach (var block in Cfg.Blocks)
             {
-                var (tree, model) = TestHelper.Compile(code, false, AnalyzerLanguage.CSharp);
-                var node = tree.GetRoot().DescendantNodes().First(x => x.RawKind == (int)syntaxKind);
-                Cfg = node.CreateCfg(model, default);
-                Lva = new RoslynLiveVariableAnalysis(Cfg, default);
-            }
-
-            public void ValidateAllEmpty()
-            {
-                foreach (var block in Cfg.Blocks)
-                {
-                    Validate(block, null, Array.Empty<Expected>());
-                }
+                Validate(block, null, []);
             }
+        }
 
-            public void ValidateEntry(params Expected[] expected) =>
-                Validate(Cfg.EntryBlock, null, expected);
+        public void ValidateEntry(params Expected[] expected) =>
+            Validate(Cfg.EntryBlock, null, expected);
 
-            public void ValidateExit(params Expected[] expected)
-            {
-                expected.All(x => x.Kind == ExpectedKind.Captured).Should().BeTrue("Exit block should expect only Captured variables.");
-                Validate(Cfg.ExitBlock, null, expected);
-            }
+        public void ValidateExit(params Expected[] expected)
+        {
+            Array.TrueForAll(expected, x => x.Kind == ExpectedKind.Captured).Should().BeTrue("Exit block should expect only Captured variables.");
```

Comment:
Use `expected.Should().AllSatisfy(...` instead

Reply:
Issue is that expected for some tests is empty, and the `AllSatisfy` is failing if the collection is empty.
Maybe I should divide these tests (with empty and non empty expected) - wdyt?